sematic/resolvers/resource_requirements.py

```diff
@@ -49,6 +50,150 @@ class KubernetesSecretMount:
+
+    Options
+    -------
+    NoSchedule:
+        The toleration indicates that the pod can run on the node even
+        if it has specified a NoSchedule taint, assuming the rest of
+        the toleration matches the taint.
+    PreferNoSchedule:
+        The toleration indicates that the pod can run on the node even
+        if it has specified a PreferNoSchedule taint, assuming the rest
+        of the toleration matches the taint.
+    NoExecute:
+        The pod will not be evicted from the node even if the node has
+        specified a NoExecute taint, assuming the rest of the toleration
+        matches the taint.
+    All:
+        The pod will not be evicted from the node even if the node has
+        any kind of taint, assuming the rest of the toleration
+        matches the taint.
+    """
+
+    NoSchedule = "NoSchedule"
+    PreferNoSchedule = "PreferNoSchedule"
+    NoExecute = "NoExecute"
+    All = "All"
+
+
+@dataclass
+class KubernetesToleration:
+    """Toleration for a node taint, enabling the pod for the function to run on the node
+
+    See Kubernetes documentation for more:
+    https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+
+    Attributes
+    ----------
+    key:
+        The key for the node taint intended to be tolerated. If empty, means
+        to match all keys AND all values
+    operator:
+        The way to compare the key/value pair to the node taint's key/value pair
+        to see if the toleration applies
+    effect:
+        The effect of the node taint the toleration is intended to tolerate.
+        Leaving it empty means to tolerate all effects.
+    value:
+        If the operator is Equals, this value will be compared to the value
+        on the node taint to see if this toleration applies.
+    toleration_seconds:
+        Only specified when effect is NoExecute (otherwise is an error). It
+        specifies the amount of time the pod can continue executing on a node
+        with a NoExecute taint
+    """
+
+    key: Optional[str] = None
+    operator: KubernetesTolerationOperator = KubernetesTolerationOperator.Equal
+    effect: KubernetesTolerationEffect = KubernetesTolerationEffect.All
+    value: Optional[str] = None
+    toleration_seconds: Optional[int] = None
+
+    def to_api_keyword_args(self) -> Dict[str, Optional[Union[str, int]]]:
+        """Convert to the format for kwargs the API python client API for tolerations"""
+        effect: Optional[str] = self.effect.value
+        if self.effect == KubernetesTolerationEffect.All:
+            # the actual API makes "all" the default behavior with no other way to
+            # specify
+            effect = None
+        operator = self.operator.value
+        return dict(
+            effect=effect,
+            key=self.key,
+            operator=operator,
+            toleration_seconds=self.toleration_seconds,
+            value=self.value,
+        )
+
+    def assert_valid(self):
```

Comment:
Thoughts on doing this at instantiation?

Reply:
moved to `__post_init__`. TIL!